chalicelib/checks/higlass_checks.py

```diff
@@ -197,71 +197,49 @@ def check_files_for_higlass_viewconf(connection, **kwargs):
-    check.status = 'PASS'
-
-    if not target_files_by_ga:
-        # nothing new to generate
-        check.summary = check.description = "No new view configs to generate"
-    else:
-        all_files = sum([len(target_files_by_ga[ga]) for ga in target_files_by_ga])
-        check.summary = "Ready to generate %s Higlass view configs" % all_files
-        check.description = check.summary + ". See full_output for details."
-        check.allow_action = True
-    return check
-
-@action_function(file_accession=None)
-def patch_files_for_higlass_viewconf(connection, **kwargs):
-    """ Action that is used with generate_higlass_view_confs_files to actually
-    POST new higlass view configs and PATCH the old files.
+    # If there are no new files to create or the user does not want to patch new files, we can report here.
+    if not (target_files_by_ga and kwargs['patch_files']):
+        check.full_output = {
+            'reference_files': reference_files_by_ga,
+            'target_files': target_files_by_ga,
+        }
+        check.status = 'PASS'
 
-    Args:
-        connection: The connection to Fourfront.
-        **kwargs, which may include:
-            file_accession(string, optional, default=None): Only generate a viewconf for the given file acccession.
+        if not target_files_by_ga:
+            check.summary = check.description = "No new view configs to generate"
+        elif not kwargs['patch_files']:
+            all_files = sum([len(target_files_by_ga[ga]) for ga in target_files_by_ga])
+            check.summary = "Ready to generate %s Higlass view configs" % all_files
+            check.description = check.summary + ". See full_output for details. Run with patch_files=True to create Higlass viewconfs and patch Files."
+        return check
 
-    Returns:
-        A check/action object.
-    """
-    action = init_action_res(connection, 'patch_files_for_higlass_viewconf')
+    # Check is complete. Now it's time to POST new higlass view configs and PATCH the files.
     action_logs = {
         'new_view_confs_by_file': {},
         'failed_to_create_viewconf' : {},
         'failed_to_patch_file' : {},
     }
 
-    # get latest results
-    gen_check = init_check_res(connection, 'check_files_for_higlass_viewconf')
-    if kwargs.get('called_by', None):
-        gen_check_result = gen_check.get_result_by_uuid(kwargs['called_by'])
-    else:
-        gen_check_result = gen_check.get_primary_result()
-
     # make the fourfront auth key (in basic auth format)
     ff_auth = (connection.ff_keys['key'], connection.ff_keys['secret'])
     headers = {'Content-Type': 'application/json',
                'Accept': 'application/json'}
 
-    # pointer to the reference files (by genome_assembly)
-    ref_files_by_ga = gen_check_result['full_output'].get('reference_files', {})
-
     # Checks expire after 280 seconds, so keep track of how long this task has lasted.
     start_time = time.time()
     time_expired = False
```

Comment:
`start_time` is defined earlier; shouldn't be here (I'm guessing it is just a remnant from the action)